plugins/device/device.proto

```diff
@@ -0,0 +1,108 @@
+syntax = "proto3";
+package hashicorp.nomad.plugins.device;
+
+import "google/protobuf/empty.proto";
+
+// DevicePlugin is the API exposed by device plugins
+service DevicePlugin {
+  // FingerprintDevices allows the device plugin to return a set of   
+  // detected devices and provide a mechanism to update the state of
+  // the device.
+  rpc FingerprintDevices(google.protobuf.Empty) returns (stream DetectedDevices) {}
+
+  // Allocate is called by the client before starting an allocation
+  // that requires access to the plugin’s devices. The plugin can use
+  // this to run any setup steps and provides the mounting details to 
+  // the Nomad client
+  rpc Allocate(AllocateRequest) returns (AllocateResponse) {}
+}
+
+// DetectedDevices is the set of devices that the device plugin has 
+// detected and is exposing
+message DetectedDevices {
+  // device_name is the name of the device. It should be scoped by 
+  // vendor  (“nvidia/gpu”, “intel/gpu”, “amd/gpu”, …)
+  string device_name = 1;
+  
+  // devices is the set of devices detected by the plugin.
+  repeated DetectedDevice devices = 2;
+
+  // node_attributes allows adding node attributes to be used for 
+  // constraints or affinities.
+  map<string, string> node_attributes = 3;
```

Comment:
what's the key in this map? 

Would it be better to push the notion of a map of attributes down a level into `DetectedDevice` instead? The `pci_bus_id` could be one of those attributes, rather than a separate field. That feels more natural to me than having the response have a list of detected devices, and a separate map of attributes from all those devices.

Reply:
The key is completely opaque to Nomad but the pci_bus_id is not so that is why we don't want to intermix the two. The node attributes aren't specific to each device,  so they are stored at the higher level